pkg/kubectl/cmd/patch.go

```diff
@@ -67,7 +67,7 @@ func NewCmdPatch(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 			cmdutil.CheckErr(err)
 		},
 	}
-	cmd.Flags().StringP("patch", "p", "", "The patch to be applied to the resource JSON file.")
+	cmd.Flags().StringP("patch", "p", "", "The patch to be applied to the resource JSON file, or - to use files on STDIN as the patch source.")
```

Comment:
I commented on this in the issue:
https://github.com/kubernetes/kubernetes/issues/17028#issuecomment-155697264